prow/cmd/deck/tide.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/tide"
+)
+
+type tideAgent struct {
+	log  *logrus.Entry
+	path string
+
+	sync.Mutex
+	pools []tide.Pool
+}
+
+func (ta *tideAgent) start() {
+	if err := ta.update(); err != nil {
+		ta.log.WithError(err).Error("Updating pool for the first time.")
+	}
+	go func() {
+		for range time.Tick(time.Minute) {
+			if err := ta.update(); err != nil {
+				ta.log.WithError(err).Error("Updating pool.")
+			}
+		}
+	}()
+}
+
+func (ta *tideAgent) update() error {
+	var pools []tide.Pool
+	var resp *http.Response
+	var err error
+	for i := 0; i < 3; i++ {
+		if err != nil {
+			ta.log.WithError(err).Warning("Tide request failed. Retrying.")
+			time.Sleep(5 * time.Second)
+		}
+		resp, err = http.Get(ta.path)
+		if err == nil {
+			defer resp.Body.Close()
+			if err := json.NewDecoder(resp.Body).Decode(&pools); err != nil {
+				return err
+			}
+			break
+		}
+	}
```

Comment:
Do we want to overwrite `ta.pools` in case we fail all 3 times? Also, can you log that last error, too?

Reply:
Fixed to return that error.